recursive_clean.sh

```
@@ -0,0 +1,67 @@
+processed_branches=()
+
+while IFS= read -r current_base_branch <<< "${branches_to_process[0]}"; do
+  branches_to_process=("${branches_to_process[@]:1}") # Remove the processed branch
+
+  # Find branches merged into the current base branch
+  merged_branches=$(git branch --merged "$current_base_branch" | grep -v "$(echo "$current_base_branch" | sed 's/$/$/')")
+
+  # Delete the merged branches
+  echo "Deleting branches merged into '$current_base_branch':"
+  echo "$merged_branches" | while read -r branch_to_delete; do
+    if [[ ! " ${processed_branches[@]} " =~ " ${branch_to_delete} " ]]; then
```

Shellcheck (0.10.0)

[error] Arrays implicitly concatenate in [[ ]]. Use a loop (or explicit * instead of @).

(SC2199)